examples/2d/mesh2d_manual.rs

```diff
@@ -30,9 +30,9 @@ use bevy::{
         Extract, Render, RenderApp, RenderSet,
```

Comment:
This example is crashing with a wgpu validation error (and it wasn't on main).

Reply:
So I did some testing, and the issue here seems to be that the format used for the vertex colors conflicts with the default `ColorMaterial`, and crashes the app, causing this rather obscure error. This does happen on `main` as well: if you give the mesh a `ColorMaterial`, it crashes, but if you remove the vertex colors, it works again.

I think there's two easy options to fix the example here: make `ColorMesh` require `HasMaterial2d` to prevent the default material from being used, or change the vertex color format to the default format `[f32; 4]`. I chose the latter here, since I think it's slightly less questionable.

However, it is still pretty problematic that the default material seems to break non-default vertex color formats (and maybe some other attributes?). It should probably handle these conflicts more gracefully, but I'm not sure how exactly. There should also definitely be a clearer error message here, but that's probably worth its own ticket.

Reply:
With the current fix, does that mean we're rendering _both_ with the default material _and_ the custom logic in this PR?

If so, I think thats a strong indicator that we should add the `HasMaterial2d` required component.

Reply:
Yeah, you're right, I added `HasMaterial2d` as a required component for it now. Although this of course only applies to this example, and doesn't really address the general problems with using vertex colors with other materials, especially default ones.